Editorialize variability considerations (#16)
intarchboard/draft-protocol-greasing: draft-edm-protocol-greasing.md

# Considerations for Increasing Protocol Variability {#variability}

> Greasing can maintain protocol extensibility by falsifying active use of it's

Reviewer: ```suggestion
Greasing can maintain protocol extensibility by falsifying active use of its
```

> extensions points. However, it does not ensure positive use. A protocol may

Reviewer: ```suggestion
extension points. However, greasing alone does not ensure positive use of extension mechanisms. A protocol may
```

define a wide-ranging extension capability that remains unused in the absence of
real use cases. This can lead to ossification that does not expect extensions,
leading to interoperability problems later on.

Long-term maintenance and interoperability can be ensured by exercising
extension points positively. To some extent this can be thought of as protocol
fuzzing. This might be difficult to exercise because varying the protocol
elements might change the outcome of interactions, leading to real errors.
However, some protocols allow elements to be be safely changed, as shown in the
following examples.

## Example: QUIC frames

QUIC packets contain frames. Receivers might build expectations on the
longitudinal aspects of packets or frames - size, ordering, frequency, etc. A
sender can quite often manipulate these parameters and stay compliant to the
requirements of the QUIC protocol.

> QUIC streams are an ordered reliable byte stream that is serialized as a

Reviewer: ```suggestion
A QUIC stream is an ordered reliable byte stream that is serialized as a
```
